src/components/JoinList.jsx

```diff
@@ -0,0 +1,60 @@
+import { useState } from 'react';
+import { useNavigate } from 'react-router-dom';
+import db from '../lib/firebase';
+import { collection } from 'firebase/firestore';
+import { useCollection } from 'react-firebase-hooks/firestore';
+import { ToastContainer, toast } from 'react-toastify';
+
+export const JoinList = ({ setToken }) => {
+  const [tokenInput, setTokenInput] = useState(' ');
+
+  const notify = () => toast('Token not valid');
+  const navigate = useNavigate();
+
+  const [value] = useCollection(collection(db, tokenInput), {
+    snapshotListenOptions: { includeMetadataChanges: true },
+  });
+
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    if (value.docs.length < 1) {
+      notify();
+    } else {
+      localStorage.setItem('token', tokenInput);
+      setToken(tokenInput);
+      if (tokenInput) navigate('/list');
+    }
+  };
+
+  const handleChange = (e) => {
+    if (e.target.value !== '') {
+      const value = e.target.value.trim();
```

Comment:
I believe this trim() method on the controlled input is causing a bug where the user cannot enter whitespace while they're typing out the token. Not an issue if they copy/paste, but it may negatively impact UX if the token is being entered manually. I would suggest moving this mutation to the handleSubmit function, before storing the input in state and localStorage.

Reply:
+1

Reply:
I agree. Thanks for picking this up in testing.

Reply:
Added a sanitize helper function, that converts to lower case and strips off the extra space . Removed value property from input field (was setting it with token input which was causing issues.